addons/sale_stock/models/sale_order.py

```diff
@@ -204,7 +207,14 @@ def action_view_delivery(self):
             picking_id = picking_id[0]
         else:
             picking_id = pickings[0]
-        action['context'] = dict(action.get('context', {}), default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_group_id=picking_id.group_id.id, group_by=[])
+        action['context'] = dict(
+            safe_eval(action['context'].strip(), eval_context),
+            default_partner_id=self.partner_id.id,
+            default_picking_type_id=picking_id.picking_type_id.id,
+            default_origin=self.name,
+            default_group_id=picking_id.group_id.id,
+            group_by=[],
+        )
```

Comment:
I've had an error flare up when accessing pickings from the sale order due to the original commit that changed this context handling.
![Captura de pantalla de 2021-11-24 09-50-27](https://user-images.githubusercontent.com/47854752/143205280-1e525e0a-ad20-4ab9-aee2-52493259e7b4.png)

Using @rco-odoo's comment as a reference I made this change: 
```python
eval_ctx = dict(self.env.context)
try:
    ctx = safe_eval(action.get("context", "{}"), eval_ctx)
except:
    ctx = {}
action['context'] = dict(ctx, default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_group_id=picking_id.group_id.id, group_by=[])
```

Which fixed my issue and returned it to working as intended, but I'm unsure of the nuances if this is the correct patch for these contexts or not, since you seem to be handling all of the changed ones

Reply:
Hi @hildickethan thank you for the feedback :)